pkg/server/singleprocess/service_ui_project.go

```diff
@@ -10,6 +10,29 @@ import (
 	serverptypes "github.com/hashicorp/waypoint/pkg/server/ptypes"
 )
 
+func (s *Service) UI_ListProjects(
+	ctx context.Context,
+	req *pb.UI_ListProjectsRequest,
+) (*pb.UI_ListProjectsResponse, error) {
+	// TODO(jgwhite): Handle pagination
```

Comment:
We should also add API validations here to ensure the `req` as the expected parameters set and aren't `nil` which could cause panics in the server.